### PR TITLE
PTCD-696 Suppress Apprenticeship Frameworks

### PR DESCRIPTION
**Pull request by jameslawesynergy**
Updated ApprenticeshipsController.ViewAndEdit to show only found Apprenticeships with RecordStaus.Live
Added GetApprenticeships Cosmos query and GetAllFrameworkApprenticeshipIds Function
Added ArchiveApprenticeships Function

### CODE REVIEW

File: src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        public GetAllPublishedFrameworkApprenticeshipIds(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        {
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher ?? throw new ArgumentNullException(nameof(cosmosDbQueryDispatcher));
+        }
+
+        [FunctionName("GetAllPublishedFrameworkApprenticeshipIds")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequest req, ILogger log)
+        {
+            var results = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetApprenticeships(a => a.FrameworkId != null));
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs:26]: Need an extra condition in the predicate here to check status (or the query is mis-named); `GetApprenticeships` doesn't look to be only returning published records

Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs:26]: https://github.com/SkillsFundingAgency/dfc-coursedirectory/pull/1717/files#diff-4ecd5d0b7c870949dc17394bc2c3e3efad0b75be40cf3fe318098835a0ceef51R21?

File: src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/Queries/GetApprenticeships.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Models;
+
+namespace Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries
+{
+    public class GetApprenticeships : ICosmosDbQuery<IDictionary<Guid, Apprenticeship>>
+    {
+        public GetApprenticeships(Expression<Func<Apprenticeship, bool>> predicate = null)
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/Queries/GetApprenticeships.cs:10]: We don't typically have constructors on DbQuery classes

File: src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs (hunk 1 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs:26]: That's not the same as 'Published'

File: src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/Queries/GetApprenticeships.cs (hunk 2 above)
Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/Queries/GetApprenticeships.cs:10]: Any particular reason? It seems like the nicest place to put this.

Comment (gunndabad) [reply src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/Queries/GetApprenticeships.cs:10]: Consistency. Not super bothered; I foresee these types all becoming records soon :-)

File: src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs (hunk 1 above)
Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs:26]: Given the scope and intention of the story, I think we want to archive all Frameworks that aren't either `Archived` or `Deleted`. So I think the problem here is with the naming of the function, not the execution. What would you refer to for all Apprenticeships that are neither `Archived` or `Deleted`?

Comment (gunndabad) [reply src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs:26]: We don't really have a word for that. That said, we always exclude `Archived` and `Deleted` so for me that part is implied. I'd go with something like `GetApprenticeshipIdsWithFramework`.

**Review by gunndabad — APPROVED** [review]

File: src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs (hunk 1 above)
Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs]: I've renamed it to `GetAllFrameworkApprenticeshipIds` - Frameworks will be basically dead once this story is completed, as such so will this function.

**Pull request merged**